core/src/components/select/select.tsx

```diff
@@ -179,11 +179,18 @@ export class Select implements ComponentInterface {
       this.setFocus();
     });
 
+    await overlay.present();
+
+    // focus selected option for popovers
     if (this.interface === 'popover') {
-      await (overlay as HTMLIonPopoverElement).presentFromTrigger(event, true);
-    } else {
-      await overlay.present();
+      let indexOfSelected = this.childOpts.map(o => o.value).indexOf(this.value);
+      indexOfSelected = indexOfSelected > -1 ? indexOfSelected : 0; // default to first option if nothing selected
+      const selectedEl = this.overlay.querySelector<HTMLElement>(`.select-interface-option:nth-child(${indexOfSelected + 1})`);
```

Comment:
```suggestion
      const selectedEl = overlay.querySelector<HTMLElement>(`.select-interface-option:nth-child(${indexOfSelected + 1})`);
```

Minor perf thing, but we already have a local reference to overlay so no need to access it in the `this` scope.